connect_ext_ppr/client/ns.py

```diff
@@ -65,26 +65,31 @@ class Service(
     NSBase,
     ActionMixin,
 ):
-    def _get_service_path(self):
-        aps_type_object = self.client.execute_request(
-            method='GET',
-            path=f'{self.path}/aps/2/resources/?implementing({self.aps_type})',
-        )
-
-        if not aps_type_object:
-            raise TypeError(f'Not able to find out Service with APS Type: {self.aps_type}')
-        elif len(aps_type_object) != 1:
-            raise TypeError(f'Multiple instances found with APS Type: {self.aps_type}')
-        else:
-            service_id = aps_type_object[0]['aps']['id']
-            return f'{self.path}/aps/2/resources/{service_id}'
-
     def __init__(self, client, aps_type: str, path: str):
         super().__init__(
             client=client,
             path=path,
         )
         self.aps_type = aps_type
+        self.__service_path = None
+
+    @property
+    def service_path(self):
+        if not self.__service_path:
```

Comment:
You can reuse the `cache_property` decorator to achieve the same behaviour

https://docs.python.org/3/library/functools.html#functools.cached_property